src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -1,80 +1,114 @@
+import seedu.address.model.recipe.Name;
+import seedu.address.model.recipe.ServingSize;
+import seedu.address.model.recipe.Step;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new EditCommand object
+ */
+public class EditCommandParser implements Parser<EditCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the EditCommand
+     * and returns an EditCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public EditCommand parse(String userInput) throws ParseException {
+        requireNonNull(userInput);
+
+        ArgumentMultimap argMultimap;
+        Prefix prefix = userInput.contains(PREFIX_INDEX.getPrefix())
+                ? PREFIX_INDEX
+                : PREFIX_NAME;
+
+        if (userInput.contains(PREFIX_INDEX.getPrefix())) {
+            argMultimap = ArgumentTokenizer.tokenize(userInput, PREFIX_INDEX, PREFIX_NAME, PREFIX_COMPLETION_TIME,
+                    PREFIX_SERVING_SIZE, PREFIX_INGREDIENT, PREFIX_TAG, PREFIX_STEP);
+        } else {
+            argMultimap = ArgumentTokenizer.tokenize(userInput, PREFIX_NAME, PREFIX_COMPLETION_TIME,
+                    PREFIX_SERVING_SIZE, PREFIX_INGREDIENT, PREFIX_TAG, PREFIX_STEP);
+        }
+
+        Index index;
+        Name name;
+
+        try {
+            if (prefix.equals(PREFIX_INDEX)) {
+                index = RecipeBookParserUtil.parseIndex(argMultimap.getValue(PREFIX_INDEX).get());
+                EditRecipeDescriptor editRecipeDescriptor = parseEditArgumentValues(argMultimap);
+                return new EditCommand(index, editRecipeDescriptor);
+            } else {
+                name = RecipeBookParserUtil.parseName(argMultimap.getPreamble());
+                EditRecipeDescriptor editRecipeDescriptor = parseEditArgumentValues(argMultimap);
+                return new EditCommand(name, editRecipeDescriptor);
+            }
+        } catch (ParseException pe) {
+            String errorMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE);
+            throw new ParseException(errorMessage, pe);
+        }
+    }
+
+    private EditRecipeDescriptor parseEditArgumentValues(ArgumentMultimap argMultimap) throws ParseException {
+        EditRecipeDescriptor editRecipeDescriptor = new EditRecipeDescriptor();
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            Name parsedName = RecipeBookParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+            editRecipeDescriptor.setName(parsedName);
+        }
+
+        if (argMultimap.getValue(PREFIX_COMPLETION_TIME).isPresent()) {
+            CompletionTime parsedCompletionTime =
+                    RecipeBookParserUtil.parseCompletionTime(argMultimap.getValue(PREFIX_COMPLETION_TIME).get());
+            editRecipeDescriptor.setCompletionTime(parsedCompletionTime);
+        }
```

Comment:
Maybe extracting the different if statements into individual methods and calling them sequentially would reduce this method length and improve code quality.
```suggestion
    private EditRecipeDescriptor parseEditArgumentValues(ArgumentMultimap argMultimap) throws ParseException {
        EditRecipeDescriptor editRecipeDescriptor = new EditRecipeDescriptor();
        parseName(editRecipeDescriptor, argMultimap);
        parseCompletionTime(editRecipeDescriptor, argMultimap);
```